src/main.go

```diff
@@ -37,6 +41,7 @@ func main() {
 
 	logFile := setupLogger(logDir, logFileName)
 	defer logFile.Close()
+	setupSosExtrasReport(sosReportFolder, sosReportFile, path.Join(logDir, logFileName))
```

Comment:
I was thinking if this call shouldn't be inside the setupLogger function. From, testing POV it makes sense to have the call separately.. I am just thinking what happens in case when the setupLogger fails on creation of log folder/file? it just logs error but the execution continues, should we handle it better?

Maybe it is just too improbable and doesn't make sense to handle that at all :shrug:

Reply:
That's a good point... I was thinking the same yesterday.

Maybe it would be better to just move that call inside `setupLogger` because if everything goes right, the sosreport initialization will happen, otherwise, we won't have that file hanging on the system.